src/mgr/StandbyPyModules.cc

```diff
@@ -168,9 +168,6 @@ void *StandbyPyModules::LoadConfigThread::entry()
 bool StandbyPyModule::get_config(const std::string &key,
                                  std::string *value) const
 {
-  PyThreadState *tstate = PyEval_SaveThread();
-  PyEval_RestoreThread(tstate);
```

Comment:
@wumingqiao It still holds the GIL below by calling PyEval_RestoreThread.

You should instead drop the GIL before calling into get_config and then re-acquire it, like:
```c++
PyThreadState *tstate = PyEval_SaveThread(); // drop GIL
bool found = self->this_module->get_config(what, &value);
PyEval_RestoreThread(tstate); // re-take GIL
```

then together with this change it will make the deadlock go away

Reply:
Ah, got it! thank you for your suggestion